### DISCUSSION TITLE
Google Workspace Marketplace

### DISCUSSION THREAD

**Discussion by hschiavo** [osm-search/Nominatim#3106] [Ideas]
https://workspace.google.com/u/0/marketplace/appfinder?host=sheets

**Describe the solution you'd like**
Use a complement in google spreadsheets to transform geocoordinates into addresses (Reverse Geocoding).

**Describe alternatives you've considered**
They have a complement with this idea, which feeds with data from Google Maps.
https://workspace.google.com/u/0/marketplace/app/geocode_by_awesome_table/904124517349

**Comment by mtmail** [comment]
I converted this from an issue to a discussion. It's something the wider community would have to create and maintain, not the Nominatim team.

In the past I worked on [Geocode in Google Sheets](https://opencagedata.com/tutorials/geocode-in-googledocs) (feel free to try). Such a tool it's not a good fit to run against the public nominatim.openstreetmap.org servers though

  - One can't rate-limit the requests, Google servers are sending 10s to 100s per second in parallel
  - Google uses anonymous servers from their Google Cloud with a generic 'google sheets' user agent string
  - It's essentially bulk geocoding
  - Unclear how caching works, it sometimes refreshes the whole sheet sending requests again
Those three points would already be against the https://operations.osmfoundation.org/policies/nominatim/

Furthermore as a developer it's not easy to customize the output (lots of users want something custom), deal with configuration (e.g. ask for language preference), version control and error handling (either you open dialog boxes or print errors into the sheet). I believe the Google tools have improved in recent years, I haven't checked in a while.
